Optional JIRA task

two-legs opened the pull request:
We have a case where there is not always a task in JIRA, so it would be great to be able to make the field with JIRA optional.

polarn commented:
Sorry for being super slow.. May I ask why you use this module if you don't have any Jira issue? :)

two-legs commented:
@polarn We have a JIRA task in most cases, but we have exceptions. These may be some very small changes that we don't create tasks for.

polarn commented:
Allright understand. And you don't change the default behaviour so it should be fine! Can you rebase it and make it into one single commit? Because I see my and Julius commits in the git log of this branch?

two-legs commented:
Done

polarn commented:
Then I try it I get this:
```
?  (optional) (DAZ-5642)
```

And before it was like:
```
? Enter JIRA issue (DAZ-12345) (DAZ-5642)
```

Haven't had time to check the code much, do you get it working correctly when you test it? Seems like the message created is not correct...

two-legs commented:
Looks like the formatter broke the code after fix... 
I fixed and added tests for dynamic questions. Also I fixed similar bug with the scope question.

polarn commented on index.js:
```diff
@@ -47,7 +47,12 @@ const options = {
+    defaults.maxLineWidth,
+  jiraOptional: getEnvOrConfig(
+    process.env.CZ_JIRA_OPTIONAL,
+    config.jiraMode,
```
@two-legs this line and the next you need to replace `jiraMode` with `jiraOptional` 😉

two-legs replied:
Fixed

polarn reviewed (changes requested):
Classic copy and paste miss. 😄

polarn commented:
Thanks for contributing! Next time I'll try to be faster...

polarn commented:
:tada: This PR is included in version 6.2.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@digitalroute/cz-conventional-changelog-for-jira)
- [GitHub release](https://github.com/digitalroute/cz-conventional-changelog-for-jira/releases/tag/v6.2.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
